src/components/i18n/i18n.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import i18n from 'i18next';
+import XHR from 'i18next-xhr-backend';
+import { initReactI18next } from 'react-i18next';
+
+const i18nInit = lng =>
+  i18n
+    .use(XHR)
+    .use(initReactI18next)
+    .init({
+      backend: {
+        loadPath: process.env.REACT_APP_CONFIG_SERVICE_LOCALES_PATH
+      },
+      fallbackLng: process.env.REACT_APP_CONFIG_SERVICE_LOCALES_DEFAULT_LNG,
+      lng,
+      debug: process.env.REACT_APP_ENV !== 'production',
+      ns: ['default'],
+      defaultNS: 'default',
+      react: {
+        useSuspense: false
+      }
+    });
+
+/**
+ * FixMe: the "I18nextProvider" appears to have timing issues, reevaluate this alt implementation
+ */
+const I18n = ({ children, locale }) => {
+  i18nInit(locale);
```

Comment:
Do we know if this is getting called multiple times (if app re-renders)? Should it or should we track if it has already been initialized?

Reply:
Yeahhhhh, was noticing some speed differences here. Went back and double checked we should probably change it to be more in line with a normal life cycle method.
### before
What happens currently during the firing...
- Initial app view load it fires twice, thats expected, the second time being the fulfilled locales ajax call
- View load, it'll fire once
- Subsequent navigation, page loads, modal edits do not affect it since the views aren't being reloaded
### after the below change
What happens after we switch over to the below
- Initial app view load it fires twice, as expected when the props refresh from the ajax call, that's it, no subsequent calls on view change... also I feel silly for missing that, I'll make the change =)
```
class I18n extends React.Component {
  componentDidMount() {
    const { locale } = this.props;

    i18nInit(locale);
  }

  componentDidUpdate(prevProps) {
    const { locale } = this.props;

    if (locale !== null && locale !== prevProps.locale) {
      i18nInit(locale);
    }
  }

  render() {
    const { children } = this.props;
    return <React.Fragment>{children}</React.Fragment>;
  }
}
```